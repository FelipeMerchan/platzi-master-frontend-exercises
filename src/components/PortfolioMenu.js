import React, { useEffect } from 'react'
import { Link } from 'react-router-dom'

import './styles/PortfolioMenu.css'

function PortfolioMenu(props) {
  const composeEvent = new Event('composed')

  const handleClick = event => {
    event.preventDefault()
    props.onClick(event)
    event.target.dispatchEvent(composeEvent);
  }

  useEffect(() => {
    const links = document.querySelectorAll('.PortfolioMenu__item > a')

    links.forEach(object => {
      object.addEventListener('composed', function(event) {
        event.stopPropagation()
        console.log('IÂ´m not propagating')
      })
    })
  })

  return (
    <nav className="PortfolioMenu">
      <ul>
        <li className="PortfolioMenu__item">
          <a role="button" onClick={ (event) => handleClick(event) } data-button={props.dataset} to="#" id="home">inicio</a>
        </li>
        <li className="PortfolioMenu__item">
          <a role="button" onClick={ (event) => handleClick(event) } data-button={props.dataset} to="#" id="aboutMe">acerca</a>
        </li>
        <li className="PortfolioMenu__item">
          <a role="button" onClick={ (event) => handleClick(event) } data-button={props.dataset} to="#" id="portfolio">portafolio</a>
        </li>
        <li className="PortfolioMenu__item">
          <a role="button" onClick={ (event) => handleClick(event) } data-button={props.dataset} to="#" id="contact">contacto</a>
        </li>
      </ul>
    </nav>
  )
}

export default PortfolioMenu