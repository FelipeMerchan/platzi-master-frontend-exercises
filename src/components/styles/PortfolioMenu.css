.PortfolioMenu {
  display: inline-block;
  height: 60px;
  line-height: 60px;
  font-family: var(--font-4);
  font-weight: var(--extra-bold);
  --darkColor: #001023;
}

.PortfolioMenu ul {
  display: flex;
}

.PortfolioMenu__item {
  margin-inline-end: var(--space-4);
  list-style: none;
}

.PortfolioMenu__item>a {
  display: inline-block;
  height: inherit;
  color: var(--darkColor);
}