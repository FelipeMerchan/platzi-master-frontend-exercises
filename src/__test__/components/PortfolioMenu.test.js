import React from 'react'
import { shallow, mount } from 'enzyme'
import { create } from 'react-test-renderer'
import '../../setupTest'
import PortfolioMenu from '../../components/PortfolioMenu'

describe('Probar el componente <PortfolioMenu />', () => {
  const portfolioMenu = shallow(<PortfolioMenu />)
  test('Render del componente PortfolioMenu', () => {
    expect(portfolioMenu.length).toEqual(1)
  })
  test('Render de los 4 botones del menú', () => {
    expect(portfolioMenu.find('ul').children()).toHaveLength(4);
  });
})

describe('Probar los comportamientos del componente PortfolioMenu', () => {
  const onClick = jest.fn()
  const portfolioMenu = mount(<PortfolioMenu onClick={onClick} />, { attachTo: document.body })
  const homeButton = portfolioMenu.find('#home')
  test('Comprobar que los botones del footer cambien de color a través de los clics de los botones de navegación', () => {
    homeButton.simulate('click')
    expect(onClick).toHaveBeenCalledTimes(1);
  });
})

describe('PortfolioMenu Snapshot', () => {
  const portfolioMenu = create(<PortfolioMenu />)
  test('Comprobar la UI del componente PortfolioMenu', () => {
    expect(portfolioMenu.toJSON()).toMatchSnapshot()
  })
})