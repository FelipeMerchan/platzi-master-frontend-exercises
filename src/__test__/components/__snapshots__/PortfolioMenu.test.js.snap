// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PortfolioMenu Snapshot Comprobar la UI del componente PortfolioMenu 1`] = `
<nav
  className="PortfolioMenu"
>
  <ul>
    <li
      className="PortfolioMenu__item"
    >
      <a
        id="home"
        onClick={[Function]}
        role="button"
        to="#"
      >
        inicio
      </a>
    </li>
    <li
      className="PortfolioMenu__item"
    >
      <a
        id="aboutMe"
        onClick={[Function]}
        role="button"
        to="#"
      >
        acerca
      </a>
    </li>
    <li
      className="PortfolioMenu__item"
    >
      <a
        id="portfolio"
        onClick={[Function]}
        role="button"
        to="#"
      >
        portafolio
      </a>
    </li>
    <li
      className="PortfolioMenu__item"
    >
      <a
        id="contact"
        onClick={[Function]}
        role="button"
        to="#"
      >
        contacto
      </a>
    </li>
  </ul>
</nav>
`;
